### PR TITLE
For #44951, login dropdown fix

### PR DESCRIPTION
**Pull request by jfboismenu**
Fixes 3 small issues Francis found during QA.
- The new combo box widgets weren't grayed out when renewing session
- Constant crash on exit on Windows 7 (not 10!) in the Desktop
- When switching back and forth between version of core the UI wasn't consistent.

### CODE REVIEW

File: python/tank/authentication/login_dialog.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def __init__(
         # show or hide the login and password fields.
         self.ui.site.lineEdit().textEdited.connect(self._site_url_changing)
         # If a site has been selected, we need to update the login field.
-        self.ui.site.activated.connect(lambda x: self._on_site_changed())
+        self.ui.site.activated.connect(self._on_site_changed)
```
Comment (jfboismenu) [inline python/tank/authentication/login_dialog.py]: Fixes the crash on exit.

File: python/tank/authentication/login_dialog.py [hunk 2]
```diff
@@ -422,9 +422,11 @@ def _ok_pressed(self):
         """
         # Wait for any ongoing SSO check thread.
         QtGui.QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)
-        if not self._query_task.wait(THREAD_WAIT_TIMEOUT_MS):
-            logger.warning("Timed out awaiting check for SSO support on the site: %s" % self._get_current_site())
-        QtGui.QApplication.restoreOverrideCursor()
+        try:
+            if not self._query_task.wait(THREAD_WAIT_TIMEOUT_MS):
```
Comment (jfboismenu) [inline python/tank/authentication/login_dialog.py]: Just in case...

File: python/tank/authentication/resources/login_dialog.ui [hunk 3]
```diff
@@ -71,6 +71,20 @@ QComboBox:down-arrow {
 QLineEdit:Disabled {
     background-color: rgb(60, 60, 60);
     color: rgb(160, 160, 160);
+}
+
+QComboBox::drop-down:disabled {
```
Comment (jfboismenu) [inline python/tank/authentication/resources/login_dialog.ui]: Fixes the styling when deactivated.

File: python/tank/authentication/session_cache.py [hunk 4]
```diff
@@ -462,6 +472,31 @@ def get_current_host():
...
+    """
+    # Extract the list of recent items.
+    items = document[recent_field]
+
+    # Then check if the current is part of the list. It's not? This is because
+    # an older core updated the file, but didn't know about the recent list and
+    # didn't update it. This is possible because since day one the authentication.yml
+    # has been treated as document with certain fields set and when the document
+    # is rewritten it is rewritten as is, except for the bits that were updated.
+    if document[current_field] and document[current_field] not in items:
```
Comment (jfboismenu) [inline python/tank/authentication/session_cache.py]: Ensures  changes back and forth of core version keep the content of the login dialog coherent.

File: python/tank/authentication/session_cache.py [hunk 5]
```diff
@@ -205,6 +205,11 @@ def _try_load_site_authentication_file(file_path):
     :returns: site authentication style dictionary
     """
     content = _try_load_yaml_file(file_path)
+
+    # No not attempt to filter out content that is not understood. This allows
```
Comment (josh-t) [inline python/tank/authentication/session_cache.py]: **Do** not... ?

File: python/tank/authentication/session_cache.py [hunk 6]
```diff
@@ -228,6 +233,11 @@ def _try_load_global_authentication_file(file_path):
     :returns: global authentication style dictionary
     """
     content = _try_load_yaml_file(file_path)
+
+    # No not attempt to filter out content that is not understood. This allows
```
Comment (josh-t) [inline python/tank/authentication/session_cache.py]: **Do** not... ?

File: python/tank/authentication/session_cache.py [hunk 7]
```diff
@@ -462,6 +472,31 @@ def get_current_host():
...
+
+    If the recent_field is not set, then we'll simply return the current_field's
+    value. The recent_field will be empty when upgrading from an older core
+    that didn't support the recent users/hosts list.
+
+    :param object document: Document to extract information from
+    :param recent_field: Field from which we need to retrieve
+    """
+    # Extract the list of recent items.
+    items = document[recent_field]
```
Comment (josh-t) [inline python/tank/authentication/session_cache.py]: for extra safety, maybe `items = document.get(recent_field)`? Is this a dictionary?

File: python/tank/authentication/session_cache.py (hunk 4 above)
Comment (josh-t) [reply python/tank/authentication/session_cache.py]: Maybe use `.get` for all of these for extra safety?

File: python/tank/authentication/session_cache.py (hunk 7 above)
Comment (jfboismenu) [reply python/tank/authentication/session_cache.py]: Not necessary. The code that reads the file actually fills in missing fields, if any, so the code afterwards can just assuming is properly formatted.

**Pull request closed without merging**